XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/Hands/BaseHandController.cs

```diff
@@ -0,0 +1,236 @@
+
+                    newHandBounds.Encapsulate(kvp.Value.Position);
+                }
+
+                if (bounds.ContainsKey(TrackedHandBounds.Hand))
+                {
+                    bounds[TrackedHandBounds.Hand] = newHandBounds;
+                }
+                else
+                {
+                    bounds.Add(TrackedHandBounds.Hand, newHandBounds);
+                }
+            }
+
+            // TrackedHandBounds.IndexFinger
+            if (JointPoses.TryGetValue(TrackedHandJoint.IndexKnuckle, out MixedRealityPose indexKnucklePose)
+                && JointPoses.TryGetValue(TrackedHandJoint.IndexMiddleJoint, out MixedRealityPose indexMiddlePose))
+            {
+                Bounds newIndexFingerBounds = new Bounds(indexKnucklePose.Position, Vector3.zero);
+                newIndexFingerBounds.Encapsulate(indexMiddlePose.Position);
+
+                if (bounds.ContainsKey(TrackedHandBounds.IndexFinger))
+                {
+                    bounds[TrackedHandBounds.IndexFinger] = newIndexFingerBounds;
+                }
+                else
+                {
+                    bounds.Add(TrackedHandBounds.IndexFinger, newIndexFingerBounds);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Updates the controller's velocity / angular velocity.
+        /// </summary>
+        protected void UpdateVelocity()
+        {
+            if (frameOn == 0)
+            {
+                deltaTimeStart = Time.unscaledTime;
+                lastPosition = GetJointPosition(TrackedHandJoint.Palm);
+                lastPalmNormal = PalmNormal;
+            }
+            else if (frameOn == velocityUpdateInterval)
+            {
+                //update linear velocity
+                float deltaTime = Time.unscaledTime - deltaTimeStart;
+                Vector3 newVelocity = (GetJointPosition(TrackedHandJoint.Palm) - lastPosition) / deltaTime;
+                Velocity = (Velocity * currentVelocityWeight) + (newVelocity * newVelocityWeight);
+
+                //update angular velocity
+                Vector3 currentPalmNormal = PalmNormal;
+                Quaternion rotation = Quaternion.FromToRotation(lastPalmNormal, currentPalmNormal);
+                Vector3 rotationRate = rotation.eulerAngles * Mathf.Deg2Rad;
+                AngularVelocity = rotationRate / deltaTime;
+            }
+
+            frameOn++;
+            frameOn = frameOn > velocityUpdateInterval ? 0 : frameOn;
+        }
+
+        /// <inheritdoc />
+        public bool TryGetBounds(TrackedHandBounds handBounds, out Bounds? bounds)
+        {
+            if (this.bounds.ContainsKey(handBounds))
+            {
+                bounds = this.bounds[handBounds];
+                return true;
+            }
+
+            bounds = null;
+            return false;
+        }
+
+        /// <inheritdoc />
+        public override void SetupDefaultInteractions(Handedness controllerHandedness)
```

Comment:
Is this section correct?  You've created left and right-handed defaults but then only use the Defaults?  Did you mean to put a left/right check in here, or are the two-handed defaults superfluous